### PR TITLE
Make Ansible-based automation work

### PR DESCRIPTION
**Pull request by nbraud**
I think this is the minimum we needed for Ansible to be in a useable state:
- [x] passes FQDNs to SSH for connecting (previously used a custom `ssh_config` that hardcoded things about @equinox0815's environment);
- [x] dropped decommissioned machines/VMs & currently-not-manageable ones;
- [x] Set host variables so deploying the `base` role everywhere does not break access for non-NOC users (esp. on `ctf`, `athsdisc` and `wuerfel`).

Note that this involved reconfiguring all machines to allow SSHing as `root`, setting keys, ...

Things that are _not_ done in this PR:
- [ ] Creating users for NOC members and allowing them SSH access.
- [x] Generating `ssh_config` snippets from templates (as per the discussion with @equinox0815)

### CODE REVIEW

File: ansible/host_vars/ctf [hunk 1]
```diff
@@ -0,0 +1,3 @@
+sshd_allowusers_set:
```
Comment (equinox0815) [inline ansible/host_vars/ctf]: This should be `sshd_allowusers_host`.

File: ansible/host_vars/wuerfel [hunk 2]
```diff
@@ -0,0 +1 @@
+sshd_allowgroup: ssh
```
Comment (equinox0815) [inline ansible/host_vars/wuerfel]: I would call the variable(s) `sshd_allowgroups_host` and `sshd_allowgroups_group`. As with `sshd_allowusers_*` the first will be set using host host_vars and the latter using group_vars.

File: ansible/roles/base/tasks/main.yaml [hunk 3]
```diff
@@ -1,25 +1,52 @@
...
+- name: limit allowed users (1/3)
   lineinfile:
      dest: /etc/ssh/sshd_config
-     regexp: "^AllowUsers"
-     line: "AllowUsers {{ ' '.join([ 'root' ] | union(sshd_allowusers_group | default([])) | union(sshd_allowusers_host | default([]))) }}"
-  when: "{{ sshd_allowusers_set | default(true) }}"
+     regexp: "^#?AllowUsers"
+     line: "AllowUsers {{ ' '.join(sshd_allowusers) }}"
+  when: sshd_allowusers_set is defined and sshd_allowgroup is not defined
   notify: restart ssh
```
Comment (equinox0815) [inline ansible/roles/base/tasks/main.yaml]: Do we need `sshd_allowusers_set`? This variable was/is needed as a workaround for one host @ Radio Helsinki and has imho no need @ realraum.

I would use:
`when: (sshd_allowusers | length) > 0 and (sshd_allowgroups | length) == 0`

File: ansible/roles/base/tasks/main.yaml [hunk 4]
```diff
@@ -1,25 +1,52 @@
...
+
+- name: limit allowed users (3/3)
   lineinfile:
      dest: /etc/ssh/sshd_config
-     regexp: "^AllowUsers"
+     regexp: "^Allow(Users|Groups)"
      state: absent
-  when: "not {{ sshd_allowusers_set | default(true) }}"
+  when: sshd_allowusers_set is not defined and sshd_allowgroup is not defined
   notify: restart ssh
```
Comment (equinox0815) [inline ansible/roles/base/tasks/main.yaml]: Same here: `sshd_allowusers_set` is not needed for our purposes.

File: ansible/roles/base/tasks/main.yaml [hunk 5]
```diff
@@ -1,25 +1,52 @@
 ---
+- set_fact:
+    sshd_allowusers: >-
+      {{ [ 'root' ] | union(sshd_allowusers_group | default([]))
+                    | union(sshd_allowusers_host  | default([])) }}
```
Comment (equinox0815) [inline ansible/roles/base/tasks/main.yaml]: sshd_allowgroups = `[] | union(sshd_allowgroups_host  | default([])) | union (sshd_allowgroups_group | default([]))`

Comment (nbraud) [reply ansible/roles/base/tasks/main.yaml]: Automatically adding users to `sshd_allowgroup` becomes a bit tricky if we want to support multiple groups, though.

**Comment by equinox0815** [comment]
lgtm

File: ansible/roles/localconfig/tasks/main.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  copy:
+    src: ssh/
+    dest: ~/.ssh/config.d/
+
+- name: assemble ssh config
+  assemble:
+    src: ~/.ssh/config.d/
+    regexp: "\\.conf$"
+    dest: ~/.ssh/config
+  when: localconfig_overwrite_ssh_config == True
```
Comment (nbraud) [inline ansible/roles/localconfig/tasks/main.yml]: Should be just `when: localconfig_overwrite_ssh_config`

**Pull request merged**